File: api/src/main/java/com/sinergy/chronosync/dto/request/AppointmentTypeRequestDTO.java
package com.sinergy.chronosync.dto.request;

import com.sinergy.chronosync.model.Firm;
import com.sinergy.chronosync.model.appointmentType.AppointmentType;
import com.sinergy.chronosync.model.appointmentType.Currency;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for creating or updating an appointment type.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AppointmentTypeRequestDTO {

	private Long id;
	private String name;
	private Integer durationMinutes;
	private Double price;
	private Currency currency;
	private String colorCode;
	private Firm firm;

	/**
	 * Creates and returns AppointmentType model from Data Transfer Object.
	 *
	 * @return {@link AppointmentType} user model
	 */
	public AppointmentType toModel() {
		return AppointmentType.builder()
			.id(id)
			.name(name)
			.durationMinutes(durationMinutes)
			.price(price)
			.currency(currency)
			.colorCode(colorCode)
			.firm(firm)
			.build();
	}
}

File: gui/src/dtos/requests/AppointmentTypeRequestDTO.ts
import { Currency } from '../../models/appointmentType/Currency';

/**
 * Appointment type request DTO.
 *
 * @property {number} id - The ID of the appointment type
 * @property {string} name - The name of the appointment type
 * @property {number} durationMinutes - The duration of the appointment type in minutes
 * @property {number} price - The price of the appointment type
 * @property {string} colorCode - The color code of the appointment type (hex code)
 * @property {Currency} currency - The currency of the appointment type
 */
export interface AppointmentTypeRequestDTO {
  id: number;
  name: string;
  durationMinutes: number;
  price: number;
  colorCode: string;
  currency: Currency;
}
